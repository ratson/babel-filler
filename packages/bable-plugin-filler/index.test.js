'use strict'
const { transform } = require('@babel/core')

const plugin = require('.')

test('fill import from default export', () => {
  const opts = {
    babelrc: false,
    plugins: [
      [
        plugin,
        {
          vars: {
            _: 'lodash',
          },
        },
      ],
    ],
  }
  const { code } = transform('_.random()', opts)
  expect(code).toMatchSnapshot()
})

test('fill import from named export', () => {
  const opts = {
    babelrc: false,
    plugins: [
      [
        plugin,
        {
          vars: {
            random: { name: 'random', module: 'lodash' },
          },
        },
      ],
    ],
  }
  const { code } = transform('random()', opts)
  expect(code).toMatchSnapshot()
})

test('fill import from named export as alias', () => {
  const opts = {
    babelrc: false,
    plugins: [
      [
        plugin,
        {
          vars: {
            f: { name: 'random', module: 'lodash' },
          },
        },
      ],
    ],
  }
  const { code } = transform('f()', opts)
  expect(code).toMatchSnapshot()
})

test('fill import from default and named export', () => {
  const opts = {
    babelrc: false,
    plugins: [
      [
        plugin,
        {
          vars: {
            Component: { name: 'Component', module: 'react' },
            React: 'react',
          },
        },
      ],
    ],
  }
  const { code } = transform(
    "class App extends Component { render() { return React.createElement('div') } }",
    opts
  )
  expect(code).toMatchSnapshot()
})

test('do nothing when no missing variables', () => {
  const opts = {
    babelrc: false,
    plugins: [
      [
        plugin,
        {
          vars: {
            _: 'lodash',
          },
        },
      ],
    ],
  }
  const { code } = transform('const _ = {}; _.random()', opts)
  expect(code).not.toContain('lodash')
})
