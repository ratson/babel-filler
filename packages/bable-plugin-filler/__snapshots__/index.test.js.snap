// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fill import from default and named export 1`] = `
"import { Component as _Component } from \\"react\\";
import _default from \\"react\\";

class App extends _Component {
  render() {
    return _default.createElement('div');
  }

}"
`;

exports[`fill import from default export 1`] = `
"import _default from \\"lodash\\";

_default.random();"
`;

exports[`fill import from named export 1`] = `
"import { random as _random } from \\"lodash\\";

_random();"
`;

exports[`fill import from named export as alias 1`] = `
"import { random as _f } from \\"source\\";

_f();"
`;
